core/src/main/java/fr/sncf/osrd/api/InfraManager.java

```diff
@@ -204,9 +205,9 @@ public FullInfra load(String infraId, String expectedVersion, DiagnosticRecorder
                 // try downloading the infra again if:
                 //  - the existing cache entry hasn't reached a stable state
                 //  - we don't have the right version
-                var obsoleteVersion = expectedVersion != null && !expectedVersion.equals(cacheEntry.expectedVersion);
+                var obsoleteVersion = expectedVersion != null && !expectedVersion.equals(cacheEntry.version);
                 if (!cacheEntry.status.isStable || obsoleteVersion)
-                    return downloadInfra(cacheEntry, infraId, expectedVersion, diagnosticRecorder);
+                    return downloadInfra(cacheEntry, infraId, diagnosticRecorder);
```

Comment:
If we don't have the right version, we download *a* version... Which may or may not be the right one... This is weird

Reply:
you're indeed correct. it might be better to retry, or even fail if the infrastructure has changed version since.

Reply:
after discussing the matter with @Khoyo, we concluded that it's part of a larger bug: https://github.com/DGEXSolutions/osrd/issues/3878